### ISSUE TITLE
Formatting header/footer text doesn't work any longer

### ISSUE THREAD

**Issue by Jojo-Schmitz** [musescore/MuseScore#24254]
### Issue type

Engraving bug (incorrect score rendering)

### Description with steps to reproduce

1. Take the score from https://musescore.org/en/node/367489
2. See the header:
    ![image](https://github.com/user-attachments/assets/5f7344ac-cbba-494e-a247-1f0e233b2c32)
3. Check Format > Style > Header/Footer and see:
     ```xml
      <font size="18"/><font face="Arial"/><b><i>$:workNumber:</i></b>
     ```
4. Open the same score with 4.3.2 and see:
    ![image](https://github.com/user-attachments/assets/2801f89c-2516-4a11-ae61-ae4fc5420c08)

### Supporting files, videos and screenshots

see above

### What is the latest version of MuseScore Studio where this issue is present?

4.4/master

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows 11, but probably all others too

### Additional context

n/a

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by r-jasper** [comment]
This is originally reported on MacOS Ventura 13.6.9 (22G830)

**Comment by mathesoncalum** [comment]
Definitely caused by one of my recent header/footer PRs (#23475, #23903, etc). Looking into it.

**Comment by r-jasper** [comment]
Resolved in version 4.4.2
